File: server/pb/groupshare/data/share.def.go
package data

import (
	"github.com/dave/groupshare/server/pb/groupshare/data/foo"
	"github.com/dave/protod/delta"
)

type Share_type struct {
	location []*delta.Locator
}

func (b Share_type) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type(l []*delta.Locator) Share_type {
	return Share_type{location: l}
}

type Share_type_repeated struct {
	location []*delta.Locator
}

func (b Share_type_repeated) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_repeated(l []*delta.Locator) Share_type_repeated {
	return Share_type_repeated{location: l}
}
func (b Share_type_repeated) Index(i int) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Index{Index: int64(i)}}))
}
func (b Share_type_repeated) Index64(i int64) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Index{Index: i}}))
}

type Share_type_bool_map struct {
	location []*delta.Locator
}

func (b Share_type_bool_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_bool_map(l []*delta.Locator) Share_type_bool_map {
	return Share_type_bool_map{location: l}
}
func (b Share_type_bool_map) Key(k bool) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Bool{Bool: k}}}}))
}

type Share_type_int32_map struct {
	location []*delta.Locator
}

func (b Share_type_int32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_int32_map(l []*delta.Locator) Share_type_int32_map {
	return Share_type_int32_map{location: l}
}
func (b Share_type_int32_map) Key(k int) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Int32{Int32: int32(k)}}}}))
}
func (b Share_type_int32_map) Key32(k int32) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Int32{Int32: k}}}}))
}

type Share_type_int64_map struct {
	location []*delta.Locator
}

func (b Share_type_int64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_int64_map(l []*delta.Locator) Share_type_int64_map {
	return Share_type_int64_map{location: l}
}
func (b Share_type_int64_map) Key(k int) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Int64{Int64: int64(k)}}}}))
}
func (b Share_type_int64_map) Key64(k int64) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Int64{Int64: k}}}}))
}

type Share_type_uint32_map struct {
	location []*delta.Locator
}

func (b Share_type_uint32_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_uint32_map(l []*delta.Locator) Share_type_uint32_map {
	return Share_type_uint32_map{location: l}
}
func (b Share_type_uint32_map) Key(k int) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Uint32{Uint32: uint32(k)}}}}))
}
func (b Share_type_uint32_map) Key32(k uint32) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Uint32{Uint32: k}}}}))
}

type Share_type_uint64_map struct {
	location []*delta.Locator
}

func (b Share_type_uint64_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_uint64_map(l []*delta.Locator) Share_type_uint64_map {
	return Share_type_uint64_map{location: l}
}
func (b Share_type_uint64_map) Key(k int) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Uint64{Uint64: uint64(k)}}}}))
}
func (b Share_type_uint64_map) Key64(k uint64) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_Uint64{Uint64: k}}}}))
}

type Share_type_string_map struct {
	location []*delta.Locator
}

func (b Share_type_string_map) Location_get() []*delta.Locator {
	return b.location
}
func NewShare_type_string_map(l []*delta.Locator) Share_type_string_map {
	return Share_type_string_map{location: l}
}
func (b Share_type_string_map) Key(k string) Share_type {
	return NewShare_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Key{Key: &delta.Key{V: &delta.Key_String_{String_: k}}}}))
}
func ShareDef() Share_type {
	return Share_type{}
}
func (b Share_type) Id() delta.String_scalar {
	return delta.NewString_scalar(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Field{Field: &delta.Field{
		Name:   "id",
		Number: 1,
	}}}))
}
func (b Share_type) Name() delta.String_scalar {
	return delta.NewString_scalar(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Field{Field: &delta.Field{
		Name:   "name",
		Number: 2,
	}}}))
}
func (b Share_type) Foo() foo.Foo_type {
	return foo.NewFoo_type(delta.CopyAndAppend(b.location, &delta.Locator{V: &delta.Locator_Field{Field: &delta.Field{
		Name:   "foo",
		Number: 3,
	}}}))
}

File: lib/pb/groupshare/data/share.def.dart
import 'package:protod/delta.dart' as delta;
import 'package:protod/delta.pb.dart' as pb;
import 'package:fixnum/fixnum.dart' as fixnum;
import 'package:groupshare/pb/groupshare/data/foo/foo.def.dart' as groupshare_data_foo_foo;

Share_type ShareDef() {
  return Share_type([]);
}

class Share_type extends delta.Location {
  Share_type(List<pb.Locator> location) : super(location);
  delta.String_scalar Id() {
    return delta.String_scalar([...location]..add(pb.Locator()..field_1 = (pb.Field()..name = "id"..number = 1)));
  }
  delta.String_scalar Name() {
    return delta.String_scalar([...location]..add(pb.Locator()..field_1 = (pb.Field()..name = "name"..number = 2)));
  }
  groupshare_data_foo_foo.Foo_type Foo() {
    return groupshare_data_foo_foo.Foo_type([...location]..add(pb.Locator()..field_1 = (pb.Field()..name = "foo"..number = 3)));
  }
}

class Share_type_repeated extends delta.Location {
  Share_type_repeated(List<pb.Locator> location) : super(location);
  Share_type Index(int i) {
    return Share_type([...location]..add(pb.Locator()..index = fixnum.Int64(i)));
  }
  Share_type Index64(fixnum.Int64 i) {
    return Share_type([...location]..add(pb.Locator()..index = i));
  }
}

class Share_type_bool_map extends delta.Location {
  Share_type_bool_map(List<pb.Locator> location) : super(location);
  Share_type Key(bool k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..bool_1 = k)));
  }
}

class Share_type_int32_map extends delta.Location {
  Share_type_int32_map(List<pb.Locator> location) : super(location);
  Share_type Key(int k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..int32 = k)));
  }
}

class Share_type_int64_map extends delta.Location {
  Share_type_int64_map(List<pb.Locator> location) : super(location);
  Share_type Key(int k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..int64 = fixnum.Int64(k))));
  }
  Share_type Key64(fixnum.Int64 k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..int64 = k)));
  }
}

class Share_type_uint32_map extends delta.Location {
  Share_type_uint32_map(List<pb.Locator> location) : super(location);
  Share_type Key(int k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..uint32 = k)));
  }
}

class Share_type_uint64_map extends delta.Location {
  Share_type_uint64_map(List<pb.Locator> location) : super(location);
  Share_type Key(int k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..uint64 = fixnum.Int64(k))));
  }
  Share_type Key64(fixnum.Int64 k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..uint64 = k)));
  }
}

class Share_type_string_map extends delta.Location {
  Share_type_string_map(List<pb.Locator> location) : super(location);
  Share_type Key(String k) {
    return Share_type([...location]..add(pb.Locator()..key = (pb.Key()..string = k)));
  }
}

